src/cache/cache_las.c

```diff
@@ -0,0 +1,390 @@
+/*-
+ * Copyright (c) 2014-2015 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_las_stats_update --
+ *	Update the lookaside table statistics for return to the application.
+ */
+void
+__wt_las_stats_update(WT_SESSION_IMPL *session)
+{
+	WT_CONNECTION_IMPL *conn;
+	WT_CONNECTION_STATS **cstats;
+	WT_DSRC_STATS **dstats;
+
+	conn = S2C(session);
+
+	/*
+	 * Lookaside table statistics are copied from the underlying lookaside
+	 * table data-source statistics. If there's no lookaside table, values
+	 * remain 0. In the current system, there's always a lookaside table,
+	 * but there's no reason not to be cautious.
+	 */
+	if (conn->las_cursor == NULL)
+		return;
+
+	/*
+	 * KEITH: is there a cleaner way to get a reference to the underlying
+	 * data handle?
+	 */
+	cstats = conn->stats;
+	dstats = ((WT_CURSOR_BTREE *)conn->las_cursor)->btree->dhandle->stats;
```

Comment:
We could cache the LAS btree handle in the connection, if that helps?  Or do a `__wt_session_get_btree` for the LAS?